src/openrct2/actions/StaffHireNewAction.hpp

```diff
@@ -264,48 +261,65 @@ DEFINE_GAME_ACTION(StaffHireNewAction, GAME_COMMAND_HIRE_NEW_STAFF_MEMBER, Staff
     void AutoPositionNewStaff(Peep * newPeep) const
     {
         // Find a location to place new staff member
-
         newPeep->state = PEEP_STATE_FALLING;
 
-        int16_t x, y, z;
         uint32_t count = 0;
-        uint16_t sprite_index;
-        Peep* guest = nullptr;
         PathElement* guest_tile = nullptr;
 
         // Count number of walking guests
-        FOR_ALL_GUESTS (sprite_index, guest)
         {
-            if (guest->state == PEEP_STATE_WALKING)
+            auto list = EntityList<Guest>(SPRITE_LIST_PEEP);
+            for (auto guest : list)
             {
-                // Check the walking guest's tile. Only count them if they're on a path tile.
-                guest_tile = map_get_path_element_at(TileCoordsXYZ{ guest->NextLoc });
-                if (guest_tile != nullptr)
-                    ++count;
+                if (guest->state == PEEP_STATE_WALKING)
+                {
+                    // Check the walking guest's tile. Only count them if they're on a path tile.
+                    guest_tile = map_get_path_element_at(TileCoordsXYZ{ guest->NextLoc });
+                    if (guest_tile != nullptr)
+                        ++count;
+                }
             }
         }
 
+        int16_t x, y, z;
```

Comment:
Could you add a TODO note here to remind us to refactor it to use (Tile)CoordsXYZ?